README.md

```diff
@@ -51,29 +51,28 @@ Here’s a complete `package.json` example, for a hypothetical module named `@mo
   "type": "module",
   "main": "./dist/index.js",
   "exports": {
-    ".": "./src/moment.mjs",
-    "./": "./src/util/",
-    "./timezones/": "./data/timezones/",
-    "./timezones/utc": "./data/timezones/utc/index.mjs"
+    "/": "./src/util/",
+    "/timezones/": "./data/timezones/",
+    "/timezones/utc": "./data/timezones/utc/index.mjs"
```

Comment:
This is already a valid absolute path on the user’s filesystem - this does not disambiguate a node_modules path.

Reply:
There's no ambiguity here IMO. The key values are not arbitrary context-less specifiers indexing the user's filesystem. They are very specifically the entrypoint into *this* package.

As an example, we could even drop the `/` prefix altogether and it would still be fully unambiguous. All of these keys are descendents of the package root directory.

Reply:
We can define it such that it’s not ambiguous, but I’m talking about the confusion angle. Any string here should imo match how require works, or be something entirely distinct.

Using a leading dot or slash is reserved for relative and absolute paths, and anything that could be a bare identifier is reserved for node_modules. We need to come up with something new here, or else we’re creating a minefield of confusion and teachability.

Reply:
My argument was that we don't need to align with require-semantics.

If we do, we only have two options:
1. Mandate a `. /` prefix
2. Use the only cross-platform character that is definitely not part of a file path: the nul character `\u0000`

The second is absurd so we must reject this PR to fulfil the goal of require-compatibility.

Reply:
Sorry if this is a crazy idea, what about prefixing with the blank scope for local imports - `@/`.  I assume npm would not allow this to be used?  Technically someone could manually create `node_modules/@/...`, not sure if that's a concern?  If so `@/` could be defined as doing local imports only when package.json contains the `exports` key.

```js
import utc from '@/timezones/utc';
```

Reply:
There has been discussion around using `~/` as a local package specifier in packages, where that would then effectively run through the "exports" system. This allows us to get internal mappings without needing to disambiguate them, but does not support private internal mappings or internal mappings of bare specifiers (eg remapping `fs` through package.json config). Perhaps that would be ok though and that is the alternative I see if we cannot get either both "exports" and "imports" or "exports" with an internal remapping disambiguation.

Reply:
@robpalme i don’t think the options are binary; but yes, i think using `~` universally would be a great solution here.

However note that if we want to restrict this to package-level relative paths only, we could certainly follow “main” and make the leading ./ optional - it’s only the possibility of supporting bare identifiers here that makes a leading ./ required.

Reply:
Personally I like using `~` to refer to physical package root for use in `require` & 'import` & as the value of an exports physical mapping.

It doesn't really make sense to use `~` as the key of an exports mapping, because that scenario is defining a virtual entrypoint, not referring to a physical location.

I think we should also be clear that virtual entrypoints (exports) are a different thing than aliases for dependencies (imports) and therefore we should not confuse them inside one pkg.json field. So the requirement to make the exports keys themselves disambiguate this fact is unnecessary.